tools/mo/openvino/tools/mo/main.py

```diff
@@ -281,13 +286,22 @@ def raise_ie_not_found():
 
     log.debug("Placeholder shapes : {}".format(argv.placeholder_shapes))
 
-    if hasattr(argv, 'extensions') and argv.extensions and argv.extensions != '':
-        extensions = argv.extensions.split(',')
-    else:
-        extensions = None
-
     argv.freeze_placeholder_with_value, argv.input = get_freeze_placeholder_values(argv.input,
                                                                                    argv.freeze_placeholder_with_value)
+
+    load_extensions(argv)
+
+    return argv
+
+
+def load_extensions(argv : argparse.Namespace):
+    moc_front_end, available_moc_front_ends = get_moc_frontends(argv)
+
+    is_tf, is_caffe, is_mxnet, is_kaldi, is_onnx =\
+        deduce_framework_by_namespace(argv) if not moc_front_end else [False, False, False, False, False]
```

Comment:
Nit: Similar code should exists somewhere outside `load_extensions`. And it indeed exists at https://github.com/openvinotoolkit/openvino/pull/9034/files#diff-5c3563b31bdd39b460b6e4fbe9c30a3737d407038e0abea8358d6bf4550a66b5R140. Not a big deal, but why do we need to detect it twice when the first time should be enough? Can we save it somehow and reuse? The original logic in this line and implied code below is not perfect and now we have duplicated it. I see your point in isolation to a separate function but it is not a reason to duplicate the code of FW detection. With new FEs we introduce some canonical names for FW in OpenVINO space, can we detect name of FW once and the build the logic based on this detection.

Reply:
@slyalin fixed in https://github.com/openvinotoolkit/openvino/pull/9034/commits/2d0c78cd7ebe3c94207fd2eb349c1daeb3f28757